lib/mongoid/gitifield.rb

```diff
@@ -27,8 +27,9 @@ def gitifields_on(fields = [])
           define_method(GITIFIELD_WORKSPACE_KEY % [field]) do
             variable_name = (GITIFIELD_WORKSPACE_CACHE_KEY % [field]).to_sym
             workspace = instance_variable_get(variable_name)
+            file_name = self.try(:name)
             if workspace.nil?
-              workspace = Workspace.new(data: self[(GITIFIELD_DATA_KEY % [field]).to_sym])
+              workspace = Workspace.new(data: self[(GITIFIELD_DATA_KEY % [field]).to_sym], file_name)
```

Comment:
`def initialize(data: '', file_name: nil, folder_name: nil)`
this should be named argument isn't it?